tests/dag_processing/test_processor.py

```diff
@@ -643,6 +643,30 @@ def test_execute_on_failure_callbacks(self, mock_ti_handle_failure):
                 test_mode=conf.getboolean('core', 'unit_test_mode'),
             )
 
+    def test_failure_callbacks_should_not_drop_hostname(self):
+        dagbag = DagBag(dag_folder="/dev/null", include_examples=True, read_dags_from_db=False)
+        dag_file_processor = DagFileProcessor(dag_ids=[], log=mock.MagicMock())
+        dag_file_processor.UNIT_TEST_MODE = False
+
+        with create_session() as session:
+            session.query(TaskInstance).delete()
```

Comment:
```suggestion
```
We have `clear_db_runs` in the setup method which clears TaskInstance as well, so we don't need this

Reply:
Sure, thank you.